src/components/CommentSection.jsx

```diff
@@ -0,0 +1,39 @@
+import { useState, useEffect } from "react";
+import { getCommentsByArticleId } from "./apiCalls";
+
+export default function CommentSection({ article_id }) {
+  const [comments, setComments] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    getCommentsByArticleId(article_id)
+      .then((data) => {
+        setComments(data.comments);
+        setIsLoading(false);
+      })
+      .catch((err) => {
+        console.error("Error getting comments:", err);
+        setError("Failed to load comments");
+        setIsLoading(false);
+      });
+  }, [article_id]);
```

Comment:
I would argue you don't need this, the article id isn't going to change unless the whole page where this is rendered reloads on a navigate.